modules/candidate_parameters/ajax/formHandler.php

```diff
@@ -477,12 +477,25 @@ function editConsentStatusFields($db, $user)
                           requires only the date of consent.');
                     return;
                 }
-            } else { // If no status stays no or record existed as NULL,
-                     // consent date and empty withdrawal date still required
+            } else { // If no status stays no or record existed as NULL
                 if (($oldStatus === null || $oldStatus === 'no') && !empty($date)
-                    && empty($withdrawal)
                 ) {
-                    $validated = true;
+                    $countYesStatusHistory = $db->pselectOne(
+                        "SELECT COUNT(*) FROM candidate_consent_history
+                        WHERE PSCID=:pscid AND ConsentName=:cn AND Status='yes'",
+                        array(
+                         'pscid' => $pscid,
+                         'cn'    => $consentName,
+                        )
+                    );
+                    // If consent was 'yes' at some point, withdrawal date is !empty
+                    // if it was never 'yes', withdrawal date should be empty
+                    // Either is valid.
+                    if ((!empty($withdrawal) && $countYesStatusHistory > 0)
+                        || (empty($withdrawal) && $countYesStatusHistory === 0)
+                    ) {
+                        $validated = true;
+                    }
```

Comment:
i believe the case you're covering above is already covered below in the else if case? its for when the 'oldStatus' = 'yes'. i'm not sure if i'm fully understanding but counting how many times in the past the consent has been 'yes' doesn't say anything about whether you're currently changing it from 'yes' to 'no'. am i missing something?

Reply:
whether date of withdrawal is set depends on whether consent was ever 'yes' in the past. 
The case you are mentioning in the else if checks whether the consent has just NOW changed from 'yes'

Reply:
you should only be putting in a date of withdrawal if you're currently withdrawing a yes consent. is this not true for you?